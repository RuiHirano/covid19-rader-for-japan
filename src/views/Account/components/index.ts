export { default as AccountDetails } from './AccountDetail';
export { default as AccountProfile } from './AccountProfile';
export { default as AccountDelete } from './AccountDelete';
