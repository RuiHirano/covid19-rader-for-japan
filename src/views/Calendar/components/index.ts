export { Details } from './Details';
export { Statistics } from './Statistics';