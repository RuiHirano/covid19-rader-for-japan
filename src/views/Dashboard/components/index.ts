export { default as Budget } from './Asset';
export { default as LatestOrders } from './LatestOrders';
export { default as LatestProducts } from './LatestProducts';
export { default as LatestCalendar } from './LatestCalendar';
export { default as LatestProfit } from './LatestProfit';
export { default as TasksProgress } from './TasksProgress';
export { default as TotalProfit } from './TotalProfit';
export { default as TotalUsers } from './ProfitRatio';
export { default as UsersByDevice } from './TradesByPair';
