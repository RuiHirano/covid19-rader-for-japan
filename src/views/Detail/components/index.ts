export { default as Content } from './Content';
export { default as DateBar } from './DateBar';