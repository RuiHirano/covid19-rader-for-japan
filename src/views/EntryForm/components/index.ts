export { BackButton } from './BackButton';
export { Form } from './Form';