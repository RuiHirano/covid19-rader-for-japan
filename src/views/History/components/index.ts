export { default as HistoryTable } from './HistoryTable';
export { default as HistoryToolbar } from './HistoryToolbar';
