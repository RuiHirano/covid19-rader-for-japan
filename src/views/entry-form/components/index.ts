export { default as BackButton } from './back-button';
export { default as Form } from './form';