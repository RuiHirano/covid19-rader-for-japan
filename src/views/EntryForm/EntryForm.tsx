import React, { useState, useEffect } from "react";
import { makeStyles, Theme } from "@material-ui/core/styles";
import { Grid, TextField, Button } from "@material-ui/core";
import * as H from "history";
import { withRouter, RouteComponentProps, match } from "react-router";
import { BackButton, Form } from "./components";
import { Formik, yupToFormErrors, FormikValues } from "formik";
import * as Yup from "yup";
import { Item, MarketType, TradeType, LoadingState } from "../../types";
import moment from "moment";
import uuid from "uuid";
import { mockItems } from "../../common/mockData";
import { useDispatch, useSelector } from "react-redux";
import { itemActions } from "../../redux/saga/Item/itemSaga";
import { useLoading } from "../../common/hooks/useLoading";
import { ItemClass } from "../../types/item";
import { AppState } from "../../redux/module/rootModule";

// Container
interface ContainerProps {
    history: H.History;
    location: H.Location;
    match: match<{ itemId: string }>;
}
const EntryFormContainer: React.FC<ContainerProps> = props => {
    const { history, match } = props;
    const dispatch = useDispatch();
    const items = useSelector((state: AppState) => state.Items);
    const handleRegistItem = (values: FormikValues) => {
        console.log("debug value", values);
        const itemClass: ItemClass = new ItemClass();
        const item: Item = {
            ID: values.ID,
            MarketType: values.MarketType,
            StartDate: values.StartDate,
            EndDate: values.EndDate,
            TradeType: values.TradeType,
            Pair: values.Pair,
            Lot: values.Lot,
            EntryRate: values.EntryRate,
            LossCutRate: values.LossCutRate,
            SettleRate: values.SettleRate,
            Profit: values.Profit,
            BeforeComment: values.BeforeComment,
            AfterComment: values.AfterComment,
            Tags: values.Tags,
            Images: values.Images,
            UpdatedAt: values.UpdatedAt,
            CreatedAt: values.CreatedAt
        };
        itemClass.setItem(item);
        if (match.params.itemId === "new") {
            dispatch(
                itemActions.createItemAction({
                    item: itemClass
                })
            );
        } else {
            dispatch(
                itemActions.updateItemAction({
                    item: itemClass
                })
            );
        }
    };

    const handleBack = () => {
        history.goBack();
    };

    const { isLoading, isFinishLoading } = useLoading(LoadingState.CREATE_ITEM);

    const [item, setItem] = useState(new ItemClass());

    useEffect(() => {
        if (isFinishLoading) {
            history.push("/dashboard");
        }
    }, [isLoading]);

    useEffect(() => {
        if (match.params.itemId === "new") {
        } else if (item.ID !== match.params.itemId) {
            items.forEach((item_: ItemClass, index: number) => {
                if (item_.ID === match.params.itemId) {
                    // urlパラメータと同じIDをもつアイテム
                    setItem(item_);
                }
            });
        }
    }, []);

    return (
        <EntryFormView
            handleRegistItem={handleRegistItem}
            item={item}
            handleBack={handleBack}
        />
    );
};

export default withRouter(EntryFormContainer);

// Presentatinal
interface Props {
    handleRegistItem: (values: FormikValues) => void;
    handleBack: () => void;
    item: ItemClass;
}

const EntryFormView: React.FC<Props> = props => {
    const { handleRegistItem, handleBack, item } = props;
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <Grid className={classes.grid} container>
                <Grid className={classes.content} item lg={12} xs={12}>
                    <div className={classes.contentHeader}>
                        <BackButton handleBack={handleBack} />
                    </div>
                    <div className={classes.contentBody}>
                        <Form item={item} handleRegistItem={handleRegistItem} />
                    </div>
                </Grid>
            </Grid>
        </div>
    );
};

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        backgroundColor: theme.palette.common.white,
        height: "100%"
    },
    contentBody: {
        flexGrow: 1,
        display: "flex",
        alignItems: "center",
        [theme.breakpoints.down("md")]: {
            justifyContent: "center"
        }
    },
    content: {
        height: "100%",
        display: "flex",
        flexDirection: "column"
    },
    contentHeader: {
        display: "flex",
        alignItems: "center",
        paddingTop: theme.spacing(5),
        paddingBototm: theme.spacing(2),
        paddingLeft: theme.spacing(2),
        paddingRight: theme.spacing(2)
    },
    grid: {
        height: "100%"
    }
}));

//export default withRouter(EntryFormView);
