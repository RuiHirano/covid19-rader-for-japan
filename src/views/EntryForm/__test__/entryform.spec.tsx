import * as React from "react";
import renderer from "react-test-renderer";
import { configure, shallow, mount } from "enzyme";
import { BackButton, Form } from "../components";
import { mockItems } from "../../../common/mockData";
import EntryForm from "../EntryForm";

describe("スクリーンショットテスト", () => {
    const item = mockItems[0];
    it("Snapshot of <Form />", () => {
        const form = shallow(
            <Form
                initialValues={{
                    ID: item.ID,
                    MarketType: item.MarketType,
                    StartDate: item.StartDate,
                    EndDate: item.EndDate,
                    TradeType: item.TradeType,
                    Pair: item.Pair,
                    Lot: item.Lot,
                    EntryRate: item.EntryRate,
                    LossCutRate: item.LossCutRate,
                    SettleRate: item.SettleRate,
                    Profit: item.Profit,
                    BeforeComment: item.BeforeComment,
                    AfterComment: item.AfterComment,
                    Tags: item.Tags,
                    Images: item.Images,
                    UpdatedAt: item.UpdatedAt,
                    CreatedAt: item.CreatedAt
                }}
            />
        );
        expect(form).toMatchSnapshot();
    });

    it("Snapshot of <BackButton />", () => {
        const backButton = shallow(<BackButton handleBack={jest.fn()} />);
        expect(backButton).toMatchSnapshot();
    });

    it("Snapshot of <BackButton />", () => {
        const entryForm = shallow(<EntryForm items={mockItems} />);
        expect(entryForm).toMatchSnapshot();
    });
});

describe("機能テスト", () => {
    it.todo("モックデータが正しく表示される");
    it.todo("newとeditで分岐ができている");
    it.todo("バックボタンを押すとアラートがでる");
    it.todo("全て入力されると登録ボタンが押せる");
    it.todo("登録ボタンを押すと登録関数が呼ばれる");
    it.todo("登録が完了したらdashboardに戻る");
});
