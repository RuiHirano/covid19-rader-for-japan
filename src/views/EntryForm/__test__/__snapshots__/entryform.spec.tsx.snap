// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スクリーンショットテスト Snapshot of <BackButton /> 1`] = `
<WithStyles(ForwardRef(IconButton))
  onClick={[MockFunction]}
>
  <ArrowBackIcon />
</WithStyles(ForwardRef(IconButton))>
`;

exports[`スクリーンショットテスト Snapshot of <BackButton /> 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`スクリーンショットテスト Snapshot of <Form /> 1`] = `
<Formik
  enableReinitialize={true}
  initialValues={
    Object {
      "AfterComment": "good",
      "BeforeComment": "good",
      "CreatedAt": "2019-11-09T15:00:00.000Z",
      "EndDate": "2019-11-09T15:00:00.000Z",
      "EntryRate": "1.0",
      "ID": "fb89d580-bb70-41e3-a3b8-3b743fb76ac6",
      "Images": Array [],
      "LossCutRate": "1.0",
      "Lot": "2.0",
      "MarketType": 0,
      "Pair": "USD/JPY",
      "Profit": "1.0",
      "SettleRate": "1.0",
      "StartDate": "2019-11-09T15:00:00.000Z",
      "Tags": Array [],
      "TradeType": 0,
      "UpdatedAt": "2019-11-09T15:00:00.000Z",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "password",
        "email",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "password": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "min": true,
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;
